x-pack/plugin/core/src/main/java/org/elasticsearch/license/ClusterStateLicenseService.java

```diff
@@ -479,6 +452,36 @@ private void updateXPackLicenseState(License license) {
         }
     }
 
+    private void maybeRegisterOrUpdateLicense(ClusterState previousClusterState, ClusterState currentClusterState) {
+        if (XPackPlugin.isReadyForXPackCustomMetadata(currentClusterState) == false) {
```

Comment:
+1 to completely remove this whole if block if you choose to ... i started to remove these across the board... but kinda forgot about it: https://github.com/elastic/elasticsearch/pull/95428

Reply:
Ah, thank you for the indication!
I also wanted to ask about it, but similarly omitted....
I've pushed [0c360c2](https://github.com/elastic/elasticsearch/pull/102919/commits/0c360c29d6f86369dce96e65c9684468be622d88) .